app/workers/import/row_worker.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class Import::RowWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: 'pull', retry: 6, dead: false
+
+  sidekiq_retries_exhausted do |msg, _exception|
+    # Increment the total number of processed items, and bump the state of the import if needed
+    bulk_import_id = BulkImportRow.where(id: msg['args'][0]).pick(:id)
```

Comment:
You need to wrap the entire block in `ActiveRecord::Base.connection_pool.with_connection`, otherwise the connection may not be released at the right time.

Reply:
Oh, I did not know about that. Fixing it.